packages/react-admin/src/mui/list/Datagrid.js

```diff
@@ -76,25 +78,43 @@ class Datagrid extends Component {
         this.props.setSort(event.currentTarget.dataset.sort);
     };
 
+    handleSelectAll = () => {
+        if (this.props.selectedIds.length > 0) {
+            return this.props.onSelect([]);
+        }
+        this.props.onSelect(this.props.ids);
+    };
+
     render() {
         const {
-            resource,
+            basePath,
+            data,
             children,
             classes,
             className,
+            currentSort,
             ids,
             isLoading,
-            data,
-            currentSort,
-            basePath,
+            resource,
             rowStyle,
+            selectedIds,
             setSort,
+            onSelect,
+            onToggleItem,
             ...rest
         } = this.props;
+
         return (
             <Table className={classnames(classes.table, className)} {...rest}>
                 <TableHead>
                     <TableRow className={classes.row}>
+                        <TableCell padding="checkbox">
```

Comment:
Couldn't the display of `<Checkbox>` (and `<DatagridRow>`) be conditioned by the addition or not of the prop `bulkActions` on `<List>` ?

Reply:
No, because we always provide a DefaultBulkActions, see the `List` component

Reply:
I agree with @alexisjanvier, we need a way to disable checkboxes on the datagrid altogether.

Reply:
Passing `false` to `bulkActions` ?